File: core/src/com/inkostilation/pong/commands/ConnectToGameCommand.java
package com.inkostilation.pong.commands;

import com.inkostilation.pong.engine.IPongEngine;

import java.io.IOException;

public class ConnectToGameCommand<M> extends AbstractPongCommand<M> {
    @Override
    public void execute(IPongEngine engine) throws IOException {
        getEngine().startNewGame(getMarker());
    }
}

File: core/src/com/inkostilation/pong/commands/ConnectToGameCommand.kt
package com.inkostilation.pong.commands

import com.inkostilation.pong.engine.IPongEngine
import java.io.IOException

class ConnectToGameCommand<M> : AbstractPongCommand<M>() {
    @Throws(IOException::class)
    override fun execute(engine: IPongEngine<*>) {
        getEngine().startNewGame(marker)
    }
}